request/http.go

```diff
@@ -0,0 +1,70 @@
+package request
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	// HTTPHeaderID is the HTTP Header to be used when sending ID in
+	// HTTP Requests or Responses
+	HTTPHeaderID = "X-REQUESTID"
+)
+
+// GetFromHTTPRequest attempts to read a Request ID from the given @r http request's
+// header. If no ID is found, or the found ID is ill-formatted, an empty ID is returned.
+func GetFromHTTPRequest(r *http.Request) ID {
+	idStr := r.Header.Get(HTTPHeaderID)
+
+	id, err := Parse(idStr)
+	if err != nil {
+		return ID{}
+	}
+
+	return id
+}
+
+// SetInHTTPRequest will add the ID registered in @ctx into the given
+// @request as a HTTP Header. If @request is nil, an warning log will
+// be emitted and nothing will be changed.
+func SetInHTTPRequest(ctx context.Context, request *http.Request) {
+	if request == nil {
```

Comment:
RequestID is never sent to the server but it's generated on the server and returned. This functions should not exist.

Reply:
I disagree that this function should not exist. The destination should never continue using the same `rid`, but there's no reason to forbid it into being sent. If anything, in some circumstances it can aid in detecting tricky situations in which `tid` can't due to its intended granularity (such as repeat attempts over the same `rid` when they shouldn't). It also aids in separating when a system generates multiples requests from a single `tid`, but a few of them are bugged but only detected in another system.

Being properly used, "requester rid" can save time. Now that we have the macro middlewares, it's very easy to add a "source_request_id" into the log for free, and no-op if the information is not available.